spanner/batch.go

```diff
@@ -305,6 +345,20 @@ func (t *BatchReadOnlyTransaction) Execute(ctx context.Context, p *Partition) *R
 				RequestOptions: p.qreq.RequestOptions,
 				ResumeToken:    resumeToken,
 			})
+			if err != nil {
+				return client, err
+			}
+			md, err := client.Header()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
nit: this early return is no longer needed now that we don't return an error if getting the latency stats fails